### PR TITLE
RESTEASY-2324: RESTEasy client should disable Apache Http Client cookie management feature by default

### PR DESCRIPTION
**Pull request by NicoNes**

### CODE REVIEW

**Comment by NicoNes** [comment]
I don' t know why but one travis build failed...

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/cdi/injection/InjectionTest.java [hunk 1]
```diff
@@ -276,7 +277,7 @@ public void testEJBs() throws Exception {
    public void testSessionScope() throws Exception {
       log.info("starting testSessionScope()");
       client.close();
-      client = ClientBuilder.newClient();
+      client = new ResteasyClientBuilderImpl().cookieManagementEnabled(true).build();
```
Comment (asoldano) [inline testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/cdi/injection/InjectionTest.java]: Please do not reference the ResteasyClientBuilderImpl directly; use:

`client = ((ResteasyClientBuilder)(ClientBuilder.newBuilder())).cookieManagementEnabled(true).build();`

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/resource/param/CookieInjectionTest.java [hunk 2]
```diff
@@ -68,7 +68,7 @@
 
    @BeforeClass
    public static void init() {
-      client = ClientBuilder.newClient();
+      client = new ResteasyClientBuilderImpl().cookieManagementEnabled(true).build();
```
Comment (asoldano) [inline testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/resource/param/CookieInjectionTest.java]: Same as before.

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/cdi/injection/InjectionTest.java (hunk 1 above)
Comment (NicoNes) [reply testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/cdi/injection/InjectionTest.java]: Done

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/resource/param/CookieInjectionTest.java (hunk 2 above)
Comment (NicoNes) [reply testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/resource/param/CookieInjectionTest.java]: Done

**Comment by NicoNes** [comment]
Hi @asoldano ,

You're right we should add documentation about this.
I tried to do so adding few words in `RESTEasy_Client_Framework.xml`. Let me know what you think about it. (And do no hesitate to correct my english also if necessary ;)).

Thanks

**Comment by asoldano** [comment]
@NicoNes sorry for the late feedback. Thanks for the addition to the doc. I'm merging this because I don't want to make the process over complicated. However, I've just realized that it would probably be better to have a ``enableCookieManagement()`` method in the builder, similarly to what we have for using async, disabling trustmanager, etc. I'll change that just after merging. Thanks.

**Comment by NicoNes** [comment]
Hey @asoldano,

I thought about it when I did it and  I hesitated because, once `enableCookieManagement()` will be called, it will not longer be possible to  build a `ResteasyClient` instance without cookie management from this builder. 
In other way the builder is no longer usable as we expect it to be.

But as you said it's already the case for other methods so let's keep it consistent.

Thanks !

**Comment by asoldano** [comment]
Follow-up on https://github.com/resteasy/Resteasy/pull/2169

**Pull request merged**